### PR TITLE
Only run unit and integration tests on oldest and latest python versions for macos

### LINKED ISSUE #405: Only run unit and integration tests on oldest and latest python versions for macos
### Problem Description

_As an engineer, I'd like to see the results of CI workflows quickly so that I cna make changes or merge PRs more efficiently._

Github only allows 5 concurrent mac machines, while it allows 20 total machines. For this reason, our CI builds often get bottlenecked at the mac builds.

### Expected behavior

- Only run the unit and integration workflows on python 3.8 and 3.12 for Macos

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86b07h928
Resolve #405

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b07h928 Copulas - Only run unit and integration tests on oldest and latest python versions for macos #405](https://app.clickup.com/t/86b07h928)

**Review by amontanez24 — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -21,15 +21,14 @@ license = { text = 'BSL-1.1' }
...
     "numpy>=1.23.3;python_version>='3.10' and python_version<'3.12'",
     "numpy>=1.26.0;python_version>='3.12'",
-    "pandas>=1.1.3;python_version<'3.10'",
-    "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
+    "pandas>=1.4.0;python_version<'3.11'",
     "pandas>=1.5.0;python_version>='3.11' and python_version<'3.12'",
     "pandas>=2.1.1;python_version>='3.12'",
     'plotly>=5.10.0',
-    "scipy>=1.5.4;python_version<'3.10'",
+    "scipy>=1.8.0;python_version<'3.10'",
```
Comment (amontanez24) [inline pyproject.toml:31]: it looks like version 1.7.3 of scipy has arm support

**Review by amontanez24 — CHANGES_REQUESTED** [review]
Just one small change

**Review by amontanez24 — APPROVED** [review]

**Review by lajohn4747 — APPROVED** [review]
Nit: Should we leave a comment about why th

**Pull request merged**